plutarch-extra/Plutarch/Extra/Map.hs

```diff
@@ -0,0 +1,335 @@
+{-# LANGUAGE QuantifiedConstraints #-}
+
+{- |
+ Module: Plutarch.Extra.Map
+ Copyright: (C) Liqwid Labs 2022
+ License: Apache 2.0
```

Comment:
? I can't accept it with this in. The license must be MIT. In addition, we don't have any header like this anywhere, albeit I'd be OK with it being shortened to a 2 lines with just the original author and sponsor.